api/api.go

```diff
@@ -31,10 +37,59 @@ var bearerRegexp = regexp.MustCompile(`^(?:B|b)earer (\S+$)`)
 
 // API is the main REST API
 type API struct {
-	handler http.Handler
-	db      *storage.Connection
-	config  *conf.GlobalConfiguration
-	version string
+	handler     http.Handler
+	db          *storage.Connection
+	config      *conf.GlobalConfiguration
+	tokenSigner *TokenSigner
+	version     string
+}
+
+type TokenSigner struct {
+	jwtConfig  *conf.JWTConfiguration
+	privateKey *rsa.PrivateKey
+}
+
+func NewTokenSigner(config *conf.Configuration) *TokenSigner {
+	t := &TokenSigner{
+		jwtConfig: &config.JWT,
+	}
+	t.init()
+	return t
+}
+
+func (t *TokenSigner) init() {
+	if t.jwtConfig.RSAPrivateKey == "" && t.jwtConfig.Algorithm == jwa.RS256.String() {
+		log.Fatal("No RSA private key configured")
+	}
+	privateKeyData, err := os.ReadFile(t.jwtConfig.RSAPrivateKey)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	block, _ := pem.Decode(privateKeyData)
```

Comment:
check the error as well?

Reply:
the `_` ignored part is not error it is remainder of bytes after reading block.

Reply:
ack